x-pack/plugins/data_quality/common/url_schema/common.ts

```diff
@@ -5,4 +5,50 @@
  * 2.0.
  */
 
+import * as rt from 'io-ts';
+
 export const DATA_QUALITY_URL_STATE_KEY = 'pageState';
+
+export const directionRT = rt.keyof({
+  asc: null,
+  desc: null,
+});
+
+export const sortRT = rt.strict({
+  field: rt.string,
+  direction: directionRT,
+});
+
+export const tableRT = rt.exact(
+  rt.partial({
+    page: rt.number,
+    rowsPerPage: rt.number,
+    sort: sortRT,
+  })
+);
+
+export const timeRangeRT = rt.strict({
+  from: rt.string,
+  to: rt.string,
+  refresh: rt.strict({
+    pause: rt.boolean,
+    value: rt.number,
+  }),
+});
+
+export const degradedFieldRT = rt.exact(
+  rt.partial({
+    table: tableRT,
+  })
+);
+
+export const dataStreamRT = new rt.Type<string, string, unknown>(
+  'dataStreamRT',
+  (input: unknown): input is string =>
+    typeof input === 'string' && (input.match(/-/g) || []).length === 2,
```

Comment:
is this for checking datastreamRT is not empty? I don't complete get it

Reply:
It check that datastreamRt is made up of `x-y-z` pattern

Reply:
Could we maybe not check this? and let it fail from the backend if that's not the case?
Reason is we are working in enabling dataset quality for other types, and probably indices that fall out of the DSNS ([reference](https://github.com/elastic/observability-design/issues/304))

Reply:
Shouldn't this be then an incremental change ? Other datasets like metrics and traces will work smooth here.
Supporting indices outside of DSNS is risky given we rely a lot on `indexNameToDataStreamParts` in multiple places. I propose we do this incrementally when we are actually doing that change with proper tests, till then, this will act as a guard. wdyt ?